test/Utils.Tests.ps1

```diff
@@ -1,6 +1,8 @@
 . $PSScriptRoot\Shared.ps1
 . $modulePath\Utils.ps1
 
+$expectedEncoding = if ($PSVersionTable.PSVersion.Major -le 5) { "utf8" } else { "ascii" }
```

Comment:
For PS Core we should default BOM-less files to `UTF8NoBOM` instead of ASCII, or maybe `Default` adequately covers both?

Reply:
There is no `Default` value for the `-Encoding` parameter on PS Core.  But obviously there is a default value that is used when `-Encoding` isn't specified.  I'm pretty sure that is utf8NoBom which is probably what we should be using on PS Core Linux/macOS.   I think UTF8 is still a good default for Windows.

Reply:
Default encoding is UTF-8 without a BOM
PowerShell Core changes the default encoding to conform with the broader ecosystems.

See https://github.com/PowerShell/PowerShell-Docs/blob/staging/reference/docs-conceptual/whats-new/What-s-New-in-PowerShell-Core-60.md#default-encoding-is-utf-8-without-a-bom

Reply:
Looks like PS Core on Windows uses `utf8NoBom` as well. 

The tricky part here that where we use this we are typically adding to an existing profile.  I would consider it a bit rude to change the file encoding of someone's profile script.  I think we are fairly safe to assume that PS Core profiles will be utf8NoBom given the default.  Not so much for Windows PowerShell though.  And our function to detect file encoding is pretty dumb.  It can't tell the diff between uft8NoBom and ASCII because it only looks for a BOM (or lack thereof).

Reply:
This might work for encoding detection: https://gist.github.com/sdwheeler/777b9ec25e914bdeb0774471a149951c#file-get-fileencoding-ps1

I think Lee Holmes wrote this. I found it on poscode.org a few years back. It does a decent job with BOM-less files.

Reply:
It is probably better than what we've got but not by much.  It appears to be just looking for a BOM as well.

Reply:
Re: no `Default`, I guess i misread https://github.com/PowerShell/PowerShell/blob/4bc52d2358222084738157a08907fac32d13bd3a/src/System.Management.Automation/utils/EncodingUtils.cs#L37

If there's no BOM, you have to make an assumption; trying to detect encoding is notoriously difficult, and we'll beyond our needs here. Pre-Core the PS assumption is ANSI; in Core the assumption is UTF8, on all platforms. I don't see much reason not to align with that. If it causes a problem for folks, they should get on board with UTF8.

The Holmes code is basically equivalent to what we have here: match by BOM, otherwise default to something (UTF7, in that case).

Reply:
> in Core the assumption is UTF8

Well, UTF8 with no BOM just to be clear/pedantic.  :-)